plugins/modules/ndo_fabric_span_session.py

```diff
@@ -0,0 +1,705 @@
+        type: str
+      port:
+        description:
+        - The destination port to use for the SPAN Session.
+        - This parameter or O(destination_port.port_uuid) is required.
+        type: dict
+        suboptions:
+          node:
+            description:
+            - The Node ID to use for the SPAN Session.
+            type: int
+            required: true
+          path:
+            description:
+            - The Node Ethernet interface to use for the SPAN Session.
+            type: str
+            required: true
+  destination_port_channel:
+    description:
+    - The destination port channel configuration group.
+    - This parameter or O(destination_epg) or O(destination_port) is required when creating the SPAN Session.
+    type: dict
+    suboptions:
+      port_channel_uuid:
+        description:
+        - The UUID of the destination port channel to use for the SPAN Session.
+        - This parameter or O(destination_port_channel.port_channel) is required.
+        type: str
+      port_channel:
+        description:
+        - The destination port channel to use for the SPAN Session.
+        - This parameter or O(destination_port_channel.port_channel_uuid) is required.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - The name of the destination port channel.
+            type: str
+            required: true
+          template:
+            description:
+            - The name of the template that contains the destination port channel.
+            - This parameter or O(destination_port_channel.port_channel.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - The ID of the  that contains the destination port channel.
+            - This parameter or O(destination_port_channel.port_channel.template) is required.
+            type: str
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Fabric Monitoring Access Policy template.
+- The O(destination_epg.epg) must exist before using it with this module in your playbook.
+  Use M(cisco.mso.mso_schema_template_anp_epg) to create the EPG.
+- The O(destination_port_channel.port_channel) must exist before using it with this module in your playbook.
+  Use M(cisco.mso.ndo_port_channel_interface) to create the Fabric resource port channel interface.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.mso_schema_template_anp_epg
+- module: cisco.mso.ndo_port_channel_interface
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create EPG - Fabric SPAN Session
```

Comment:
should we change this around because we are not creating an epg
```suggestion
- name: Create Fabric SPAN Session with destination EPG
```

same applies for other examples